### DISCUSSION TITLE
Sending Message From a Background to Background

### DISCUSSION THREAD

**Discussion by TheColaber** [ScratchAddons/ScratchAddons#541] [Ideas] [answered]
So I'm working on some context menu stuff and I'm trying to enable and disable extensions through the context menu. I noticed that the options page uses
```
chrome.runtime.sendMessage("getSettingsInfo", ({ manifests, addonsEnabled, addonSettings }) => {

})
```
But when I try it through a background.js file, it gives to errors:

`Unchecked runtime.lastError: Could not establish connection. Receiving end does not exist.`

` Error handling response: TypeError: Cannot destructure property 'manifests' of 'undefined' as it is undefined.`

Does anyone know why?

**Comment by WorldLanguages** [comment, marked as answer]
You can't make a page talk to itself. Right now the code is structured in a way where addons can only be disabled/enabled when the background page receives a specific message (because the only way to toggle an addon as of now, is from the settings page), so that will need to be restructured for a usecase like this.

**Comment by TheColaber** [comment]
@WorldLanguages Ah... ok. You think maybe I can set the result as a global variable or something or should I just not mess with this?

**Comment by WorldLanguages** [comment]
I'd recommend you do not mess with this. I was planning to add toggles on the popup, not the context menu, anyway

**Reply by TheColaber** [reply]
btw for like #9 and #19 are you still going to implement those in the future?

**Comment by TheColaber** [comment]
@WorldLanguages Ok... I'll see if I can work on something else. Lmk if you need anything from me.
